flink-core/src/test/java/org/apache/flink/util/NetUtilsTest.java

```diff
@@ -49,6 +58,30 @@ public void testParseHostPortAddress() {
         assertEquals(socketAddress, NetUtils.parseHostPortAddress("foo.com:8080"));
     }
 
+    @Test
+    public void testAcceptWithoutTimeout() throws IOException {
+        // Validates that acceptWithoutTimeout suppresses all SocketTimeoutExceptions
+        ServerSocket serverSocket = mock(ServerSocket.class);
```

Comment:
Instead of relying on mockito which we very much want to avoid, we instead can create a `ServerSocket` subclass and override `accept()` accordingly.

Reply:
Good point, will do. Out of curiosity, what is the motivation for avoiding mockito? (Apologies if this is well documented already)

Reply:
We had a fair amount of maintenance overhead due to mockito in the past; instead we aim to write code in such a way that mocking isn't necessary.